zipkin4net-aspnetcore/Criteo.Profiling.Tracing.Middleware/TracingHandler.cs

```diff
@@ -10,15 +10,14 @@ public class TracingHandler : DelegatingHandler
         private readonly ITraceInjector<HttpHeaders> _injector;
         private readonly string _serviceName;
 
-        public TracingHandler(string serviceName, HttpMessageHandler httpMessageHandler = null)
-        : this(new Middleware.ZipkinHttpTraceInjector(), serviceName, httpMessageHandler)
+        public TracingHandler(string serviceName)
+        : this(new Middleware.ZipkinHttpTraceInjector(), serviceName)
         {}
 
-        internal TracingHandler(ITraceInjector<HttpHeaders> injector, string serviceName, HttpMessageHandler httpMessageHandler = null)
+        internal TracingHandler(ITraceInjector<HttpHeaders> injector, string serviceName)
         {
             _injector = injector;
             _serviceName = serviceName;
-            InnerHandler = httpMessageHandler ?? new HttpClientHandler();
```

Comment:
InnerHandler must be set in order to work. A runtime exception is otherwise launched